crates/store/src/state.rs

```diff
@@ -236,16 +243,9 @@ impl State {
         // in-memory write lock. This requires the DB update to run concurrently, so a new task is
         // spawned.
         let db = self.db.clone();
+        let block_store = self.block_store.clone();
         let handle = tokio::spawn(async move {
-            db.apply_block(
-                allow_acquire,
-                acquire_done,
-                block.header(),
-                notes,
-                block.created_nullifiers().to_vec(),
-                block.updated_accounts().to_vec(),
-            )
-            .await
+            db.apply_block(allow_acquire, acquire_done, block_store, block, notes).await
```

Comment:
Actually, this may need to happen before line 258 as I think that's when the database transaction gets committed (might be worth adding a comment there).